src/pages/index.js

```diff
@@ -129,26 +249,65 @@ function getCardElement(data) {
   cardImageElement.src = data.link;
   cardImageElement.alt = data.name;
 
+  if (localStorage.getItem(`liked-${data._id}`) === "true") {
+    cardLikeBtn.classList.add("card__like-button_liked");
+  }
+
   cardLikeBtn.addEventListener("click", () => {
-    cardLikeBtn.classList.toggle("card__like-button_liked");
+    const isLiked = cardLikeBtn.classList.contains("card__like-button_liked");
+
+    const request = isLiked ? api.unlikeCard(data._id) : api.likeCard(data._id);
+
+    request
+      .then((updatedCard) => {
+        cardLikeBtn.classList.toggle("card__like-button_liked");
+        const nowLiked = cardLikeBtn.classList.contains(
+          "card__like-button_liked"
+        );
+        localStorage.setItem(`liked-${data._id}`, nowLiked);
+      })
+      .catch(console.error);
   });
 
   cardImageElement.addEventListener("click", () => {
     openModal(previewModal);
-
     previewModalCaptionEl.textContent = data.name;
-
     previewModalImgEl.src = data.link;
     previewModalImgEl.alt = data.name;
   });
 
-  cardDeleteBtn.addEventListener("click", () => {
-    cardElement.remove();
-  });
+  cardDeleteBtn.addEventListener("click", () =>
+    handleDeleteCard(cardElement, data._id)
+  );
 
   return cardElement;
 }
 
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = document.querySelectorAll(".modal__submit-btn");
+
+  setButtonText(deleteConfirmationBtn, true, "Deleting...");
+
+  api
+    .removeCard(selectedCardId)
+    .then(() => {
+      closeModal(deleteConfirmationModal);
+      selectedCard.remove();
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(deleteConfirmationBtn, false, "Delete");
+    });
+}
+
+function handleDeleteCard(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteConfirmationModal);
+}
+
```

Comment:
* I can't close the `confirmation` modal by clicking on the `close icon` and `cancel` button https://snipboard.io/gaSz6L.jpg